dp/core/syscall.c

```diff
@@ -199,7 +199,8 @@ static int sys_bpoll(struct bsys_desc __user *d, unsigned int nr)
 				KSTATS_PUSH(idle, NULL);
 
 				start = rdtsc();
-				eth_rx_idle_wait(deadline);
+				if (percpu_get(eth_num_queues) > 0)
```

Comment:
The correct thing to do is to check one queue. Actually checking multiple queues is redundant because every queue is assigned and polled by an individual CPU. Theoretically, some CPUs can sleep if their queue is idle while other CPUs are busy. In other words, the current implementation of `rx_idle_wait` is not appropriate.